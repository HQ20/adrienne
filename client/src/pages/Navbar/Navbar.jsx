import React from 'react';

import './Navbar.module.css';

// import avatarImg from '../../assets/avatar.svg';
import Logo from '../../assets/AdrienneLogo.png';

const Navbar = () => (
    <div className="Navbar">
        <div className="Navbar__Flex">
            <div className="Navbar__Logo">
                <img className="Navbar__Logo" src={Logo} alt="logo" />
            </div>
            <div>
                <p className="Navbar__Title inline">Hello Alberto</p>
                <p className="Navbar__Title inline">Balance : </p>
            </div>
        </div>
    </div>
);


export default Navbar;
