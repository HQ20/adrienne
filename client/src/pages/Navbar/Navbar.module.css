.Navbar {
    position: relative;
    margin-bottom: 30px;
    background: #000;
    width: 100vw;
    height: 60px;
}

.Navbar__Flex {
    display: flex;
    justify-content: space-between;
}

.Navbar__Logo {
    height: 46px;
    width: 46px;
    padding: 4px 10px;
}

.Navbar__Title {
    margin: 0px;
    color: #fff;
    font-weight: 700;
    line-height: 50px;
    padding-left: 10px;
}

.Navbar__Avatar {
    height: 40px;
    width: 40px;
    line-height: 50px;
    padding: 5px 10px;
}
