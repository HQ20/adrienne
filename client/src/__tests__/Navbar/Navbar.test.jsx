import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Web3 from 'web3';
import FakeProvider from 'web3-fake-provider';

import getWeb3 from '../../utils/getWeb3';
import Navbar from '../../pages/Navbar/Navbar';


configure({ adapter: new Adapter() });
jest.mock('../../utils/getWeb3');

const web3 = new Web3();
const provider = new FakeProvider();
web3.setProvider(provider);

test('Check navbar and it\'s values', () => {
    const fakeWeb3 = async () => web3;
    getWeb3.mockImplementation(fakeWeb3);
    const component = renderer.create(<Navbar />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
