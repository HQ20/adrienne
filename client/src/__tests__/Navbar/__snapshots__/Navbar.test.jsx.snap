// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check navbar and it's values 1`] = `
<div
  className="Navbar"
>
  <div
    className="Navbar__Flex"
  >
    <div
      className="Navbar__Logo"
    >
      <img
        alt="logo"
        className="Navbar__Logo"
        src="AdrienneLogo.png"
      />
    </div>
    <div>
      <p
        className="Navbar__Title inline"
      >
        Balance: 
      </p>
      <p
        className="Navbar__Title inline"
      >
        0
         
        EUROS
      </p>
      <p
        className="Navbar__Title inline"
      >
        Hello Alberto
      </p>
      <div
        className="inline"
      >
        <img
          alt="avatar"
          className="Navbar__Avatar"
          src="https://cdn1.iconfinder.com/data/icons/user-pictures/100/boy-512.png"
        />
      </div>
    </div>
  </div>
</div>
`;
